src/main/java/com/rocketden/main/model/User.java

```diff
@@ -18,13 +18,21 @@
 @Setter
 public class User {
 
+    // ID used for the database.
     @Id
     @GeneratedValue(strategy = GenerationType.AUTO)
     private Integer id;
 
+    // User ID for user identification and accessible from frontend.
+    @EqualsAndHashCode.Include
+    private String userId;
+    
     @EqualsAndHashCode.Include
```

Comment:
I see this as a potential second layer of verification; for example, if two users are created but not added to the database, they may not have the `userId` field, so that value could be `null` despite them both having different `nickname`s. Furthermore, if they have the same `userId`, then the should presumably in all cases have the same `nickname`, so it shouldn't change any behavior under that circumstance. (The reason I didn't include it with `sessionId` is because not all places on the frontend are updated with this value I believe, so this _could_ actually return an incorrect value.)

Thoughts? I'm still very open to changing this.